### ISSUE TITLE
Update `@yamada-ui/dialog` accessibility 

### ISSUE THREAD

**Issue by setodeve** [yamada-ui/yamada-ui#1686]
### Description
This issue aims to address accessibility deficiencies in our dialog component as per W3C guidelines. The component currently does not meet the accessibility standards outlined in the W3C documentation. The goal is to ensure the component is accessible to all users, including those using assistive technologies.

Ensure that all unchecked items in this checklist should be addressed.

#### Basic Checks
- [ ] Component has `role="dialog"`.
- [ ] Component has `aria-modal="true"`.
- [ ] `aria-labelledby` or `aria-label` is set.

#### Initial Focus
- [ ] Initial focus is set on the first focusable element inside the dialog.
- [x] Uses `tabindex="-1"` where appropriate.

#### Keyboard Interaction
- [x] Tab key moves focus to the next element inside the dialog.
- [x] Shift + Tab key moves focus to the previous element inside the dialog.
- [x] Escape key closes the dialog.

#### Focus Return
- [ ] Focus returns to the invoking element when the dialog closes.

### Problem Statement/Justification

none

### Are you willing to participate in implementing this feature and create a pull request include the implementation?

None

### References

[W3C Dialog (Modal) Pattern](https://www.w3.org/WAI/ARIA/apg/patterns/dialogmodal/)

**Comment by Tomoya-Matsubara** [comment]
I'll try this issue!

**Comment by nakayan5** [comment]
related https://github.com/yamada-ui/yamada-ui/issues/2133

**Comment by nakayan5** [comment]
@Tomoya-Matsubara 
sorry, i just noticed.

**Comment by Tomoya-Matsubara** [comment]
@nakayan5 
That's OK, I couldn't make it anyway :)
